### PR TITLE
fix: Don't collect map, flatMap, withFilter in for-comprehension

### PR DESCRIPTION
**Pull request by jkciesluk**
We don't want rename/highlight and other PC features to include occurrences in for-comp

### CODE REVIEW

File: mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala [hunk 1]
```diff
@@ -559,6 +561,14 @@ abstract class PcCollector[T](
         Span(span.start, span.start + realName.length, point)
       else Span(point, span.end, point)
     else span
+
+  // We don't want to collect synthethic `map`, `withFilter` and `flatMap` in for-comprenhensions
+  private def isForComprehensionMethod(sel: Select): Boolean =
+    val name = sel.name.decoded
+    val syntheticName =
+      name == "map" || name == "withFilter" || name == "flatMap"
```
Comment (kasiaMarek) [inline mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala:569]: Is this check necessary? Or can `wrongSpan` be true for e.g. `apply`?

**Review by kasiaMarek — APPROVED** [review]

File: mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala (hunk 1 above)
Comment (kasiaMarek) [reply mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala:569]: Also you should add "foreach".

**Review by kasiaMarek — CHANGES_REQUESTED** [review]

File: mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala (hunk 1 above)
Comment (jkciesluk) [reply mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala:569]: > Is this check necessary? Or can wrongSpan be true for e.g. apply?

Exactly , eg. in `<<ListB>>uffer(123)`, `<<>>` mark span of `apply`

Actually, I just noticed there is a bug in Scala 2:
```scala
val x = <<ListB>>uffer(1)
val y = ListBuffer.<<ap@@ply>>(2)
```
I will create an issue and look at it in a follow up

File: tests/cross/src/test/scala/tests/pc/PcRenameSuite.scala [hunk 2]
```diff
@@ -475,4 +475,17 @@ class PcRenameSuite extends BasePcRenameSuite {
        |} yield <<a@@bc>>
        |""".stripMargin,
   )
+
+  check(
+    "map-method",
```
[the remark this answers is not in the document]
Comment (jkciesluk) [reply tests/cross/src/test/scala/tests/pc/PcRenameSuite.scala:480]: Done!

**Review by kasiaMarek — APPROVED** [review]
Looks good. Just backport it to dotty and either
 1. wait a day before merging this PR
 2. add ignores for nightlies

So the tests pass on the main branch.

File: mtags/src/main/scala-2/scala/meta/internal/pc/PcCollector.scala [hunk 3]
```diff
@@ -491,4 +491,13 @@ abstract class PcCollector[T](
...
 
+  // We don't want to collect synthethic `map`, `withFilter`, `foreach` and `flatMap` in for-comprenhensions
+  private def isForComprehensionMethod(sel: Select): Boolean = {
+    val name = sel.name.decoded
+    val syntheticName =
+      name == "map" || name == "withFilter" || name == "flatMap" || name == "foreach"
+    val wrongSpan = sel.qualifier.pos.includes(sel.namePosition.focusStart)
+    syntheticName && wrongSpan
+  }
+
```
Comment (tgodzik) [inline mtags/src/main/scala-2/scala/meta/internal/pc/PcCollector.scala:502]:
```suggestion
  private val forCompMethods = Set(nme.map, nme.flatMap, nme.withFilter, nme.foreach)
  
  // We don't want to collect synthethic `map`, `withFilter`, `foreach` and `flatMap` in for-comprenhensions
  private def isForComprehensionMethod(sel: Select): Boolean = {
    val syntheticName = sel.name match {
      case name: TermName =>  forCompMethods(name)
      case _ => false
    }
    val wrongSpan = sel.qualifier.pos.includes(sel.namePosition.focusStart)
    syntheticName && wrongSpan
  }
```
and something along these lines for Scala 3?

**Review by tgodzik — CHANGES_REQUESTED** [review]
One minor thing from me

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**